test/test_learn.py

```diff
@@ -1,260 +1,78 @@
+    work_dir_path = mkdtemp()
+    try:
+        hdf5_path = os.path.join(work_dir_path, "test.hdf5")
 
-        # adress of the database
-        train_database = '1ak4.hdf5'
-        if not os.path.isfile(train_database):
-            raise FileNotFoundError(
-                'Database %s not found. Make sure to run test_generate before',
-                train_database)
+        # data_augmentation has been set to a high number, so that
+        # the train, valid and test set can be large enough.
+        data_generator = DataGenerator([mutant], data_augmentation=50,
+                                       compute_targets=target_modules,
+                                       compute_features=feature_modules,
+                                       hdf5=hdf5_path)
 
-        # clean the output dir
-        out = './out_3d_reg'
-        if os.path.isdir(out):
-            for f in glob.glob(out + '/*'):
-                os.remove(f)
-            os.removedirs(out)
+        data_generator.create_database()
 
-        # declare the dataset instance
-        data_set = DataSet(
-            train_database=train_database,
-            valid_database=None,
-            test_database=None,
-            chain1='C',
-            chain2='D',
-            mapfly=False,
-            use_rotation=2,
-            grid_shape=(30, 30, 30),
-            select_feature={
-                'AtomicDensities_ind': 'all',
-                'Feature_ind': [ 'coulomb', 'vdwaals', 'charge', 'PSSM_*']},
-            select_target='DOCKQ',
-            tqdm=True,
-            normalize_features=True,
-            normalize_targets=True,
-            clip_features=False,
-            pair_chain_feature=np.add,
-            dict_filter={ 'DOCKQ': '<1.'})
-        # dict_filter={'IRMSD':'<4. or >10.'})
+        data_generator.map_features(grid_info)
 
-        # create the networkt
-        model = NeuralNet(data_set, cnn3d, model_type='3d', task='reg',
-                          cuda=False, plot=True, outdir=out)
+        dataset = DataSet(hdf5_path, grid_info=grid_info,
+                          select_feature='all',
+                          select_target='target1',
+                          normalize_features=True,
+                          pair_chain_feature=numpy.add,
+                          dict_filter={'target1':'>=1'})
 
-        # start the training
-        model.train(
-            nepoch=5,
-            divide_trainset=0.8,
-            train_batch_size=2,
-            num_workers=0,
-            preshuffle_seed=2019,
-            save_model='all')
+        ok_(len(dataset) > 0)
+        ok_(dataset[0] is not None)
 
-    @unittest.skipIf(skip, "Torch fails on Travis")
-    @staticmethod
-    def test_learn_3d_class():
-        """Use a 3D CNN for regularization."""
+        net_output_dir_path = os.path.join(work_dir_path, 'net-output')
+        neural_net = NeuralNet(dataset, cnn_reg, model_type='3d',task='reg', save_hitrate=False,
```

Comment:
Here the 'dataset' is features associated with one variant, isn't it?

Reply:
Yes, one variant. However, I used data augmentation to make 50 orientations of the variant's structure. (lazy) All this tests is that the NeuralNet.train method doesn't throw an error. The test doesn't look at the actual results.

Do you wish to extend this test somehow?

Reply:
Since the test doesn't look at actual results, I think the existing code suffices.